srtcore/core.cpp

```diff
@@ -5774,13 +5780,25 @@ void *CUDT::tsbpd(void *param)
              */
             HLOGC(tslog.Debug, log << self->CONID() << "tsbpd: no data, scheduling wakeup at ack");
             self->m_bTsbPdAckWakeup = true;
-            THREAD_PAUSED();
-            tsbpd_cc.wait();
-            THREAD_RESUMED();
+
+            bool signaled = false;
+            while (!signaled)
+            {
+                // For safety reasons, do wakeup once per 1s and re-check the flag.
+                THREAD_PAUSED();
+                signaled = tsbpd_cc.wait_for(seconds_from(1));
+                THREAD_RESUMED();
+                if (self->m_bClosing)
+                {
+                    HLOGC(tslog.Debug, log << "tsbpd: IPE? Closing flag set in the meantime of waiting. Exiting");
+                    goto ExitLoops;
+                }
+            }
         }
 
         HLOGC(tslog.Debug, log << self->CONID() << "tsbpd: WAKE UP!!!");
     }
+ExitLoops:
```

Comment:
No new `goto` statements, please.

Reply:
I need to exit two loops, and additionally maintain the thread checker.